### DISCUSSION TITLE
Nginx Proxy Manager + Cloudflare = 404 error

### DISCUSSION THREAD

**Discussion by snuff0815** [llamafilm/tesla-http-proxy-addon#123] [Q&A]
Hi all, I can't find a way to access the the public key. Can you pls help me?

I'm using Cloudflare and Nginx Proxy Manager. I set a CNAME for telsa.mydomain.com in Cloudflare and a Proxy Host for this subdomain pointing to the IP address of my Home Assistant with the Forward Port 4430 (from the configuration of the Tesla HTTP proxy Add-on).

I've set the domain name in the add-on to tesla.mydomain.com and started the add-on. The add-on didn't stop as the readme tells. And I get the message "404: Not Found" when I test the public key endpoint.

I also tried to include a "custom location" for /.well-known/appspecific/com.tesla.3p.public-key.pem in Nginx Proxy Manager.

Nothing worked. Can you pls help me?

Here is the log from Tesla http proxy:

x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
strict-transport-security: max-age=63072000;includeSubDomains; preload
cf-cache-status: DYNAMIC
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v4?s=AOVB6LAgAZIxSAPPmOWct1fL%2BC4QelVLIdoSCqtNAx%2Bc9BX6ojgT%2FuS9iK2FfQah85Wn7ykgUv2nuB%2B%2BBVpt%2BEWIlLUp1DrArzhsgpaaFAA8ZKf%2Fb8rLTukaIe4YkZzq%2Fwdjw2bR1w%3D%3D"}],"group":"cf-nel","max_age":604800}
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 899694469c005b12-VIE
alt-svc: h3=":443"; ma=86400

HTTP/2 404 
date: Tue, 25 Jun 2024 17:11:54 GMT
content-type: text/plain; charset=utf-8
content-length: 14
curl: (22) The requested URL returned error: 404
referrer-policy: no-referrer
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
strict-transport-security: max-age=63072000;includeSubDomains; preload
Warning: Problem (retrying all errors). Will retry in 4 seconds. 12 retries 
Warning: left.
cf-cache-status: DYNAMIC
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v4?s=wr98gVfsUVewKBJ5Q07H2eWP0CUud9f0eK%2BnDscs1t6IbQ4JoggN7eGbTSykl%2FuSB1iQBm0R3UvMt8ap15nouT3lGjR87ZdcuaYOOY9%2Bo76rXXw%2F9srJP58GJKuie7RmildBGf4Chg%3D%3D"}],"group":"cf-nel","max_age":604800}
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 899694536baa5b12-VIE
alt-svc: h3=":443"; ma=86400

HTTP/2 404 
date: Tue, 25 Jun 2024 17:11:58 GMT
content-type: text/plain; charset=utf-8
content-length: 14
referrer-policy: no-referrer
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
strict-transport-security: max-age=63072000;includeSubDomains; preload
cf-cache-status: DYNAMIC
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v4?s=upV%2Frjf7Z7J%2FwhcEXw4th8ydlNb8GS010fz%2F%2BhJF8jItVlKgbLiYfmrNU1G%2FNiHzYdU82gYDtBchRgghJUbi9cob3Bfv1h4Kcr7NX%2B%2F%2Fy0KX%2FO6HJ7B%2F%2FLTzw7qpSUpjf3gVm8TycA%3D%3D"}],"group":"cf-nel","max_age":604800}
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 8996946cba955b12-VIE
alt-svc: h3=":443"; ma=86400
curl: (22) The requested URL returned error: 404
Warning: Problem (retrying all errors). Will retry in 8 seconds. 11 retries 

Warning: left.
HTTP/2 404 
date: Tue, 25 Jun 2024 17:12:06 GMT
content-type: text/plain; charset=utf-8
curl: (22) The requested URL returned error: 404
Warning: Problem (retrying all errors). Will retry in 16 seconds. 10 retries 
content-length: 14
referrer-policy: no-referrer
Warning: left.
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
strict-transport-security: max-age=63072000;includeSubDomains; preload
cf-cache-status: DYNAMIC
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v4?s=QAIWMscbmU%2BJ9se%2FiM6QtPpy4jmTcAHOXN2epxT6clAZSLR2NIpEc0lEtiGYVZ4qSyPNVxXJGmE4RzbyRaSlGc8tIENi3bbK%2BF90lifT%2BbNGYqJyyOAqEeomIjplxetiCpBJb%2BEzbw%3D%3D"}],"group":"cf-nel","max_age":604800}
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 8996949f1bed5b12-VIE
alt-svc: h3=":443"; ma=86400

HTTP/2 404 
date: Tue, 25 Jun 2024 17:12:22 GMT
content-type: text/plain; charset=utf-8
curl: (22) The requested URL returned error: 404
Warning: Problem (retrying all errors). Will retry in 32 seconds. 9 retries 
Warning: left.
content-length: 14
referrer-policy: no-referrer
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
strict-transport-security: max-age=63072000;includeSubDomains; preload
cf-cache-status: DYNAMIC
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v4?s=Zi694wBPJBwesN9jsT1%2F4lv9%2FVQ866Q4%2Bel0h8vfJOPxEp3bWEsI5EqxJekX6gjAbpNDRcTb4frvN5C5XPxPGEkZcmY%2B0rqrnLFrY1TBFMosR5%2FMvlb22PAL%2BAEDCzw0nDk4uDokxg%3D%3D"}],"group":"cf-nel","max_age":604800}
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 899695037a935b12-VIE
alt-svc: h3=":443"; ma=86400

[19:12:29] werkzeug:INFO: 172.30.32.2 - - [25/Jun/2024 19:12:29] "GET / HTTP/1.1" 200 -
HTTP/2 404 
date: Tue, 25 Jun 2024 17:12:54 GMT
content-type: text/plain; charset=utf-8
content-length: 14
curl: (22) The requested URL returned error: 404
Warning: Problem (retrying all errors). Will retry in 64 seconds. 8 retries 
referrer-policy: no-referrer
Warning: left.
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
strict-transport-security: max-age=63072000;includeSubDomains; preload
cf-cache-status: DYNAMIC
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v4?s=loAZkNAWKB%2FyeuXD4k%2FDWrUdj1jRC7CMw%2FEjsVuSxy93CPQaaViTvvbZ%2B5D6GG2AEkqKK0if5Wq8m5WeaYb2hQV0636buSY6w4zieB3BBh1ISqwe%2BefZp6YDG4C1mSki1DkIYjVp9g%3D%3D"}],"group":"cf-nel","max_age":604800}
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 899695cc0b185b12-VIE
alt-svc: h3=":443"; ma=86400

**Comment by ballonchef** [comment]
I have exactly the same issue an cannot find out how to solve it. After checking the configs and logs repeatedly, I thought it might be a certificate issue, as the internal address of homeassistant does not reply as well. So i set up new certificates etc., but still the same error here. Does "nslookup homeassistant-internal.yourdomain.com" work?
